src/Runner.Worker/Program.cs

```diff
@@ -64,5 +69,24 @@ public static async Task<int> MainAsync(IHostContext context, string[] args)
 
             return 1;
         }
+
+
+
+        /// <summary>
+        /// Runner.Worker is started by Runner.Listener in a separate process,
+        /// so the two can't be debugged in the same session.
+        /// This method halts the Runner.Worker process until a debugger is attached,
+        /// allowing a developer to debug Runner.Worker from start to finish.
+        /// </summary>
+        private static async Task WaitForDebugger(Tracing trace)
+        {
+            trace.Info($"Waiting for a debugger to be attached. Edit the 'GITHUB_ACTIONS_RUNNER_ATTACH_DEBUGGER' environment variable to toggle this feature.");
+            while (!Debugger.IsAttached)
```

Comment:
can we wait for some max time in case you don't attach a debugger

Reply:
Sure, I'm thinking 30s would probably be appropriate.